src/transformers/models/led/modeling_led.py

```diff
@@ -411,29 +411,26 @@ def _chunk(hidden_states, window_overlap, onnx_export=False):
             return hidden_states.as_strided(size=chunk_size, stride=chunk_stride)
 
         # When exporting to ONNX, use this separate logic
-        if hidden_states.size(1) == window_overlap * 2:
-            # simplest case
-            return hidden_states.unsqueeze(1)
-        else:
-            # have to use slow implementation since as_strided, unfold and 2d-tensor indexing aren't supported (yet) in ONNX export
-
-            # TODO replace this with
-            # > return hidden_states.unfold(dimension=1, size=window_overlap * 2, step=window_overlap).transpose(2, 3)
-            # once `unfold` is supported
+        # have to use slow implementation since as_strided, unfold and 2d-tensor indexing aren't supported (yet) in ONNX export
+
+        # TODO replace this with
+        # > return hidden_states.unfold(dimension=1, size=window_overlap * 2, step=window_overlap).transpose(2, 3)
+        # once `unfold` is supported
+        # the case hidden_states.size(1) == window_overlap * 2 can also simply return hidden_states.unsqueeze(1), but that's control flow
```

Comment:
So ONNX export can't never work correctly when using tensors in the control flow, if we have different tensor properties between the export time and the inference time?

Reply:
Basically onnx export follows the warning here: https://pytorch.org/docs/stable/generated/torch.jit.trace.html

If there is control flow dependent on things such as batch size / sequence length, yes indeed, the exported model will have failing cases. In case there are static attributes and only a path is recorded. For example we never enter this control flow during the export and it's totally fine: https://github.com/huggingface/transformers/blob/v4.15.0/src/transformers/models/bert/modeling_bert.py#L229

Currently the tests are inexistent in this regard: the export is done with sequence length = 8, the validation test is done with sequence length = 9. Hence the silent errors.